plugins/logicalaccess/plugins/cards/mifare/mifarenfctagcardservice.cpp

```diff
@@ -407,17 +373,27 @@ std::vector<std::shared_ptr<NfcData> > MifareNFCTagCardService::readNFC(std::sha
 
 std::shared_ptr<NdefMessage> MifareNFCTagCardService::readNDEF()
 {
-  return nullptr;
+	return std::dynamic_pointer_cast<logicalaccess::NdefMessage>(readNFC().back());
 }
 
 void MifareNFCTagCardService::writeNDEF(std::shared_ptr<NdefMessage> n)
 {
+    std::shared_ptr<logicalaccess::MifareAccessInfo> aiToWrite(
+        new logicalaccess::MifareAccessInfo());
+    aiToWrite->keyA.reset(new logicalaccess::MifareKey("d3 f7 d3 f7 d3 f7"));
```

Comment:
Same comment here, also looks like a code duplicate on read/write according to the context (inherited from original implementation?)